src/main/java/tool/Parser.java

```diff
@@ -0,0 +1,118 @@
+package tool;
+
+import task.Deadline;
+import task.Event;
+import task.Task;
+import task.Todo;
+
+
+public class Parser {
+    protected TaskList commands;
+    protected Storage storage;
+    protected Ui ui;
+
+
+    public Parser(TaskList commands, Storage storage, Ui ui) {
+        this.commands = commands;
+        this.storage = storage;
+        this.ui = ui;
+    }
+
+    /**
+     * Parses the user's input to make sense of all the commands, and hands it over
+     * to tool.TaskList
+     * @param command
+     */
+    public void parse(String command) {
+            String[] inputArr = command.split(" ");
```

Comment:
This method contains a large bulk of Duke's logic. Would it be better to split each command off into a separate method? For example:
```
if(userCommand.equals("done")) {
    try {
         int index = Integer.parseInt(inputArr[1]) - 1;
         try {
             Task doneTask = this.commands.done(index);
    /* ... */
} else //...
```

becomes

```
if(userCommand.equals("done")) {
    int index = Integer.parseInt(inputArr[1]) - 1;
    this.markTaskAsDone(index);
} else
```